test/testother.cpp

```diff
@@ -130,6 +130,7 @@ class TestOther : public TestFixture {
         TEST_CASE(duplicateBranch1); // tests extracted by http://www.viva64.com/en/b/0149/ ( Comparison between PVS-Studio and cppcheck ): Errors detected in Quake 3: Arena by PVS-Studio: Fragment 2
         TEST_CASE(duplicateBranch2); // empty macro
         TEST_CASE(duplicateBranch3);
+        TEST_CASE(duplicateBranch4);
```

Comment:
well this is a draft.. but I envision that there will be a separate PR for just updating the AST to start with.

Reply:
I can split this test into a separate PR if you like but I think it would look kind of weird. As it stands I like the fact that this PR starts with a user level bug and adds test at increasingly low levels. I think if the ast fix hadn't fixed the user level bug I wouldn't add this test in the PR but since it appears to fix the bug I think this test is implied by the title of the PR (which of course I can also change if you prefer).

Reply:
well if we can agree on some syntax soonish we can keep it here in this PR.

Reply:
ignore my comment I think I misunderstood you.